Cython/Compiler/ExprNodes.py

```diff
@@ -2919,6 +2922,24 @@ def analyse_types(self, env):
             PyrexTypes.CFuncTypeArg("it", PyrexTypes.py_object_type, None),
             ]))
 
+    def is_reversed_cpp_iteration(self):
+        """
+        Returns True if the 'reversed' function is applied to a C++ iterable.
+
+        This supports C++ classes with reverse_iterator implemented.
+        """
+        if not isinstance(self.sequence, SimpleCallNode):
```

Comment:
```suggestion
        if not (isinstance(self.sequence, SimpleCallNode) and 
                len(self.sequence.args) == 1):
```

I'm just going to add an extra check in that it matches what we expect.